### PR TITLE
11 add topic query tests and implementation

### PR DESCRIPTION
**Pull request by DominiqueCoding**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -489,4 +489,59 @@ describe("GET /api/users",()=>{
         return request(app).get("/api/userssss").expect(404)
     })
 })
+
+describe.only("GET /api/articles topic query",()=>{
```
Comment (nc-hev-mag) [inline __tests__/app.test.js]: remove .only

File: __tests__/app.test.js [hunk 2]
```diff
@@ -489,4 +489,59 @@ describe("GET /api/users",()=>{
...
+                })
+            })
+
+            
+        })
+    })
+
+    it("400: if query is invalid return error",()=>{
+        return request(app).get("/api/articles?topic=not_valid").expect(400)
+        .then(({body})=>{
```
Comment (nc-hev-mag) [inline __tests__/app.test.js:542]: 404 would suit this better for topic not found (bad request implies that the syntax was invalid)

**Review by nc-hev-mag — CHANGES_REQUESTED** [review]
Hey Dominique - great stuff here, still need a test for when a requested topic exists but has no associated articles - we'd expect an empty array!

**Review by nc-hev-mag — APPROVED** [review]
Great stuff Dominique - good to merge!
